### PR TITLE
Add FC yield in base pairs to data point

### PR DESCRIPTION
**Pull request by silverslott**
As requested by Ellen and Mattias.

It's a bit slow to load, maybe because of all the looping 🤔 not sure how to speed it up though. Could just be slow locally also.

### CODE REVIEW

**Comment by sofiahag** [comment]
To clarify- it shows when you hover a data point

**Comment by sofiahag** [comment]
I added a spinner, the loading takes 9500 ms for me though, might be unacceptable?

**Comment by sofiahag** [comment]
> Summarizing the yield might be sped up by not using core JS looping, but I don't know enough about the subject to make specific suggestions 😅
> 
> I notice on stage if you hover some data points you sometimes get a very high number of reported digits for the basepair yield such as `mbp: 26.7599999999999998` which we may want to trim down a bit 😆

I think the it might be the Python looping that is slowing it down 🤔 but _I_ don't know enough about _that_ to be certain 😏 

Yes, I noticed that too, sigh, I don't understand why the values have such different formats. Some values have both commas and dots, which I had to tend to. And now some of them won't be rounded to two decimal points 😞 I'll look into it! 👍🏻

**Comment by sofiahag** [comment]
Yes I added a timer to check how long the `def get` takes to execute:

`start_time = time.time()`
`if search_string is None:`
           ` last_month = datetime.datetime.now() - datetime.timedelta(days=30)`
            `first_term = last_month.isoformat()[2:10].replace("-", "")`
            `second_term = datetime.datetime.now().isoformat()[2:10].replace("-", "")`
`else:`
            `first_term, second_term = search_string.split('-')`

`bp_list = self.calculate_fc_yield()`
`plot_docs = [x.value for x in self.application.x_flowcells_db.view("plot/reads_yield")[first_term:second_term+'ZZZZ'].rows]`

`for k in plot_docs:`
        `for fc, bp in bp_list:`
                `if k.get('id') == fc:`
                      `k['bp_yield'] = bp`

`elapsed_time = time.time() - start_time`
        
 And it took 14 seconds 🤷🏻‍♀️

**Comment by kedhammar** [comment]
Can we use a different view or construct a smaller one? It should be possible to calculate the sums we want inside a view, instead of the backend, no?

**Comment by kedhammar** [comment]
Yeah you beat me to it 😆

**Comment by sofiahag** [comment]
> Yeah you beat me to it 😆

😆 

I used the same logic as for the reads_yield view, and it's a lot faster now! But still a couple of seconds loading time 🤷🏻‍♀️

**Comment by sofiahag** [comment]
New code up on stage!

**Comment by aanil** [comment]
Is this still WIP?

File: run_dir/static/js/fc_trend_plot.js [hunk 1]
```diff
@@ -21,12 +21,33 @@ function refresh_plot(){
...
+            events: {
+                render: function(){
+                    var formatted_sum = (sumBPYield(this.series)).toLocaleString();
+                    this.setTitle({text: 'Accumulated yield in Mbp: ' + formatted_sum}, false, false);
+                }
+            }
         },
         title: {
-            text : name+' of the recent flowcells'
+            text: function() { return name+' of the recent flowcells, yield sum in Mbp: ' +  sumBPYield(series, data, true); }
```
Comment (aanil) [inline run_dir/static/js/fc_trend_plot.js:50]: `sumBPYield(series, data, true)` only takes a single argument in the definition above?

File: run_dir/static/js/fc_trend_plot.js [hunk 2]
```diff
@@ -78,6 +99,14 @@ function make_plot(key, name, display_by, filter_inst_type, filter_inst, color_t
...
     };
 
+    function addBP() {
+        toplot.tooltip.pointFormat = '{series.name} : <b>{point.y}</b><br />Mbp: <b>{point.bp_yield:,.0f}</b>';
+    }
+
+    if (display_by == "flowcell") {
+        addBP();
+    }
+
```
Comment (aanil) [inline run_dir/static/js/fc_trend_plot.js:109]: If this is the only place addBP is used why can't this be
```suggestion
    if (display_by == "flowcell") {
        toplot.tooltip.pointFormat = '{series.name} : <b>{point.y}</b><br />Mbp: <b>{point.bp_yield:,.0f}</b>';
    }

```

**Comment by sofiahag** [comment]
> Is this still WIP?

Nope, I should have removed it long ago 😅

File: run_dir/static/js/fc_trend_plot.js (hunk 1 above)
Comment (sofiahag) [reply run_dir/static/js/fc_trend_plot.js:50]: Right, it's old, from before 👍🏻

File: run_dir/static/js/fc_trend_plot.js (hunk 2 above)
Comment (sofiahag) [reply run_dir/static/js/fc_trend_plot.js:109]: Also from old code 👍🏻

**Review by aanil — APPROVED** [review]
👍

**Pull request merged**